apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -1,5 +1,43 @@
 'use strict';
 
+var $block = null;
+
+
 $(document).ready(function() {
   // Put app logic here
-});
+  
+
+  $('[data-stack]').click(function () {
+    
+    if($block) {
+      var $lastBlock = $(this).children().last();
+      var $lastBlockValue = $lastBlock.data("block");
+      var $blockToBePlaced = $block.data("block");
+      //define variables for clean code throughout
+
+
+      if($blockToBePlaced < $lastBlockValue || !$lastBlockValue) {
+        //comparing the block that was selected, to the last block.
+        
+        $(this).append($block);
+        $block = null;
+        checkForWin();
+      }
+              
+    } else {
+      $block = $(this).children().last().detach();
```

Comment:
You could also use your predefined $lastBlock variable here if you make it accessible to the entire function scope.

`$block = $lastBlock).detach();`